packages/dev/core/src/Inputs/scene.inputManager.ts

```diff
@@ -566,7 +566,8 @@ export class InputManager {
         this._initActionManager = (act: Nullable<AbstractActionManager>): Nullable<AbstractActionManager> => {
             if (!this._meshPickProceed) {
                 const pickResult =
-                    scene.skipPointerUpPicking || (scene._registeredActions === 0 && !(scene.onPointerObservable.observers.length > this._cameraObserverCount || scene.onPointerPick || scene.onPointerUp))
+                    scene.skipPointerUpPicking ||
+                    (scene._registeredActions === 0 && !(scene.onPointerObservable.observers.length > this._cameraObserverCount || scene.onPointerPick || scene.onPointerUp))

```

Comment:
nit: Maybe factor out this logic so that we don't have to repeat it multiple times?

Reply:
Factored out